### ISSUE TITLE
README improvements - initial pass 

### ISSUE THREAD

**Issue by whowutwut** [IBM/ibm-spectrum-scale-cloud-install#9]
Attempt to clarify the README for this project and provide a better context flow.  This repository is maintained by the IBM Spectrum Scale development team, so naturally it's under the IBM org, The  "ibm-spectrum-scale" in "ibm-spectrum-scale-cloud-install" is to further help tie together various github repos that are created by the spectrum scale team. 

This repo does not actually do the work in installing IBM Spectrum Scale, but rather helps to deploy cloud infrastructure resources to AWS and Azure (with more to come in the future)

Opening this issue to help address some feedback that we have gotten from internal users of the repo, which will improve the user experience for anyone who lands here.

- [ ] Clarify the description of the project 
- [ ] Provide a more end to end flow so that the user can visualize what happens after running terraform <-- this needs some more thought. 
- [x] Clarify what are pre-requisites are vs. what the tool will actually do. 
- [x] Validate that there's a minimum clear and base success path  (Follow up with #20)

**Comment by whowutwut** [comment]
For the description of this repo
> Cloud Resource Provisioning framework for IBM Spectrum Scale

Should we just rename this to "Cloud Resource Provisioning Framework" ?

**Comment by whowutwut** [comment]
Any objections to this rename proposed above ☝️  I don't really want to open an issue to track that discussion.... a simply thumbs up/down on the above comment would indicate some agreement/disagreement...  @mamuthiah @sasikeda ?

**Comment by whowutwut** [comment]
I'm trying to also look at the README in a better flow as it's flip flopping between the two clouds we currently support...  I am consider the flowing Table of Contents Flow.  Thoughts?

-----


* Intro
* Quick Start (if any?) 
* Documentation (which links to .md files under `./docs`)
  * AWS (`./docs/aws`)
     * Template Parameters
     * Usage
  * Azure (`./docs/azure`)
     * Template Parameters
     * Usage
  * Troubleshooting (`./docs/troubleshooting`)
     * Warning
* Reporting Issues and Feedback
* Disclaimer 
* Contributing

**Comment by whowutwut** [comment]
@sasikeda  I made a bunch of updates to PR #11...   

Then I created a branch off of that to demonstrate a large refactor, as described above. I didn't create a pull request for it, but it's in my README.v1 branch: https://github.com/whowutwut/ibm-spectrum-scale-cloud-install/tree/README.v1

I'll admit that it's a pretty drastic refactor, but I feel it flows a little better and sets us up better for future cloud support.

**Comment by sasikeda** [comment]
@whowutwut 👍 I liked the simplest Table of content and it looks very clean. Its the best approach, thank you so much. As ahead up, there are going to be a few new variables introduced recently and old ones will changes. But hoping they will not trigger major changes.

**Comment by whowutwut** [comment]
@sasikeda just to clarify, are you talking about the major refactor in the [README.v1](https://github.com/whowutwut/ibm-spectrum-scale-cloud-install/blob/README.v1/README.md)  branch? 

To be more clear, I created a Work in Progress `[WIP]` pull request #24  for the major refactor.  This is built on top of PR #11 so we can pick one or the other.  I also modified the PR to target going into the  `dev` branch instead of master.
